health/health_json.c

```diff
@@ -93,18 +93,21 @@ inline void health_alarm_entry2json_nolock(BUFFER *wb, ALARM_ENTRY *ae, RRDHOST
     buffer_strcat(wb, "\t}");
 }
 
-void health_alarm_log2json(RRDHOST *host, BUFFER *wb, uint32_t after) {
+void health_alarm_log2json(RRDHOST *host, BUFFER *wb, uint32_t after, char *chart) {
     netdata_rwlock_rdlock(&host->health_log.alarm_log_rwlock);
 
     buffer_strcat(wb, "[");
 
     unsigned int max = host->health_log.max;
     unsigned int count = 0;
+    uint32_t hash_chart = 0;
+    if (chart) hash_chart = simple_hash(chart);
     ALARM_ENTRY *ae;
-    for(ae = host->health_log.alarms; ae && count < max ; count++, ae = ae->next) {
-        if(ae->unique_id > after) {
+    for(ae = host->health_log.alarms; ae && count < max ; ae = ae->next) {
+        if( ( !chart && ae->unique_id > after ) || ( chart && ae->unique_id > after && ae->hash_chart == hash_chart && !strcmp(ae->chart, chart) ) ) {
```

Comment:
Isn't is simpler with 

 (ae->unique_id > after) && (!chart || (ae->hash_chart == hash_chart && !strcmp(ae->chart, chart))))

Reply:
Yes it is!! https://github.com/netdata/netdata/pull/10788/commits/8dc332a6a5a03be7953f3c8a97a7823313dc3c59